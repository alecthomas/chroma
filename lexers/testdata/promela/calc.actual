// reverse polish

mtype = { operator, value }

chan f = [12] of { mtype, int }

proctype calc(chan you)
{	int s, lft, rgt
	chan me = [0] of { int }

	if
	:: f?operator(s)
		run calc(me); me?lft
		run calc(me); me?rgt
		if
		:: s == '+' -> you!(lft+rgt)
		fi
	:: f?value(s) -> you!s
	fi
}
