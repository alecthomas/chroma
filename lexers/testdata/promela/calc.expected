[
  {"type":"CommentSingle","value":"// reverse polish\n"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"KeywordDeclaration","value":"mtype"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"operator"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"value"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"chan"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"f"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"LiteralNumberInteger","value":"12"},
  {"type":"Punctuation","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"of"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordDeclaration","value":"mtype"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"int"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"KeywordDeclaration","value":"proctype"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"calc"},
  {"type":"Punctuation","value":"("},
  {"type":"KeywordDeclaration","value":"chan"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"you"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":"\t"},
  {"type":"KeywordType","value":"int"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"s"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"lft"},
  {"type":"Punctuation","value":","},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"rgt"},
  {"type":"TextWhitespace","value":"\n\t"},
  {"type":"KeywordDeclaration","value":"chan"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"me"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"="},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"["},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":"]"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"of"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordType","value":"int"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n\t"},
  {"type":"Keyword","value":"if"},
  {"type":"TextWhitespace","value":"\n\t"},
  {"type":"Operator","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"f"},
  {"type":"Operator","value":"?"},
  {"type":"Name","value":"operator"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"s"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n\t\t"},
  {"type":"OperatorWord","value":"run"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"calc"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"me"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"me"},
  {"type":"Operator","value":"?"},
  {"type":"Name","value":"lft"},
  {"type":"TextWhitespace","value":"\n\t\t"},
  {"type":"OperatorWord","value":"run"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"calc"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"me"},
  {"type":"Punctuation","value":");"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"me"},
  {"type":"Operator","value":"?"},
  {"type":"Name","value":"rgt"},
  {"type":"TextWhitespace","value":"\n\t\t"},
  {"type":"Keyword","value":"if"},
  {"type":"TextWhitespace","value":"\n\t\t"},
  {"type":"Operator","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"s"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"=="},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralStringChar","value":"'+'"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"you"},
  {"type":"Operator","value":"!"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"lft"},
  {"type":"Operator","value":"+"},
  {"type":"Name","value":"rgt"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":"\n\t\t"},
  {"type":"Keyword","value":"fi"},
  {"type":"TextWhitespace","value":"\n\t"},
  {"type":"Operator","value":"::"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"f"},
  {"type":"Operator","value":"?"},
  {"type":"Name","value":"value"},
  {"type":"Punctuation","value":"("},
  {"type":"Name","value":"s"},
  {"type":"Punctuation","value":")"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Operator","value":"-\u003e"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Name","value":"you"},
  {"type":"Operator","value":"!"},
  {"type":"Name","value":"s"},
  {"type":"TextWhitespace","value":"\n\t"},
  {"type":"Keyword","value":"fi"},
  {"type":"TextWhitespace","value":"\n"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n"}
]
